### PR TITLE
Make hazelcast-openshift similar to hazelcast-enterprise

### PR DESCRIPTION
**Pull request by leszko**
### Changes
- Refactor to look similar to [hazelcast-enterprise/Dockerfile](https://github.com/hazelcast/hazelcast-docker/tree/master/hazelcast-enterprise)
- Upgrade Java to 11
- Introduce features from hazelcast-enterprise: Prometheus, logging change with env variable
- Make it able to compile for Multi Arch

Note that this PR is not 100% backwards-compatible, some variables were removed. Also Java version was upgraded. But I think it should be fine, since we don't have yet any adoption for this image.

### Things to consider

- I'm not sure how much the image size is for hazelcast-enterprise users. But we could technically get rid of hazelcast-enterprise/Dockerfile and use this Dockerfile for all hazelcast-enterprise images
  - hazelcast-enterprise image size: `94MB`
  - hazelcast-openshift image size: `185MB`
- If not, we at least consider moving this repository into `hazelcast/hazelcast-docker` repository to simplify the maintenance

### CODE REVIEW

**Comment by mesutcelik** [comment]
What is the reason image size is doubled now?

> hazelcast-enterprise images
> hazelcast-enterprise image size: 94MB
> hazelcast-openshift image size: 185MB
>

**Comment by leszko** [comment]
> What is the reason image size is doubled now?
> 
> > hazelcast-enterprise images
> > hazelcast-enterprise image size: 94MB
> > hazelcast-openshift image size: 185MB

It's not now, it was always this way. Our `hazelcast-openshift` image is way bigger than our `hazelcast-enterprise` image, because UBI base image is way bigger than alpine.

**Comment by kwart** [comment]
For me is the size blocker. If the price for RedHat base image is another 100MB, I don't want to pay it in Enterprise images. Let keep the fat only in images where it's mandatory.

**Comment by leszko** [comment]
Just to be clear, I was maybe not clear enough. This PR does not change the image size, it may reduce it a little bit, but in general it does not change it.

What I compared is the current state of our Docker images. We have two Hazelcast Enterprise images:
- Docker Hub: [hazelcast/hazelcast-enterprise](https://hub.docker.com/r/hazelcast/hazelcast-enterprise) - `94MB`
- Red Hat Container Catalog: [hazelcast-enterprise-4.x](https://catalog.redhat.com/software/containers/detail/5e3d0089ac3db9037071de44) - `185MB`

We could merge them, but the size would double. I kind-of agree with Josef that doubling the size is probably a blocker.

In any case, it's not related to this PR.

File: Dockerfile [hunk 1]
```diff
@@ -1,68 +1,79 @@
...
-ENV HZ_HOME="/opt/hazelcast" \
-    HZ_CP_MOUNT="${HZ_HOME}/external" \
-    LANG="en_US.utf8" \
-    USER_NAME="hazelcast" \
-    USER_UID=10001
-
-ENV HZ_VERSION 4.0
-
+# Versions of Hazelcast and Hazelcast plugins
+ARG HZ_VERSION=4.0.1
```
Comment (hasancelik) [inline Dockerfile:5]: @lazerion and @Holmistr,
I do not have detailed info about new release script but version update mechanism might need to change, FYI. For instance, [old script ](https://github.com/hazelcast/DevOps/blob/master/build-tool/build.xml#L144) searches and modifies **ENV** not **ARG**.

**Comment by hasancelik** [comment]
> ### Things to consider
> - I'm not sure how much the image size is for hazelcast-enterprise users. But we could technically get rid of hazelcast-enterprise/Dockerfile and use this Dockerfile for all hazelcast-enterprise images
>>hazelcast-enterprise image size: `94MB`
>>hazelcast-openshift image size: `185MB`
> - If not, we at least consider moving this repository into `hazelcast/hazelcast-docker` repository to simplify the maintenance

Both options are OK for me. I think size is not huge blocker when you consider maintenance cost of multiple Dockerfiles. Also UBI(similar to RHEL) is good fit for enterprise product/customer in terms of security etc.

**Comment by leszko** [comment]
> Also UBI(similar to RHEL) is good fit for enterprise product/customer in terms of security etc.

This is also true, that you can consider UBI (supported by Red Hat) image more secure, so may fit better enterprise customers.

File: Dockerfile [hunk 2]
```diff
@@ -1,68 +1,79 @@
...
-    ./mvnw -B -f dependency-copy.xml dependency:copy-dependencies && \
-    rm -rf "$HZ_HOME/mvnw" "$HZ_HOME/.m2"
+RUN mkdir -p "${HZ_HOME}/lib" "$HZ_CP_MOUNT" \
+    && echo "Disabling subscription-manager plugin to prevent redundant logs" \
+    && sed -i 's/^enabled=.*/enabled=0/g' /etc/dnf/plugins/subscription-manager.conf \
+    && dnf config-manager --disable \
+    && dnf update -y  && rm -rf /var/cache/dnf \
+    && dnf -y update-minimal --security --sec-severity=Important --sec-severity=Critical --setopt=tsflags=nodocs \
+    && echo "Installing new packages" \
+    && dnf -y --setopt=tsflags=nodocs install java-11-openjdk apr openssl &> /dev/null \
```
Comment (kwart) [inline Dockerfile:62]: We don't need `apr` and `openssl` packages when using the `netty-tcnative-boringssl-static` lib.

File: Dockerfile (hunk 1 above)
Comment (lazerion) [reply Dockerfile:5]: Thanks @hasancelik I think this confirms your expectation regarding **ENV** update
https://github.com/hazelcast/hazelcast-pipeline-library/blob/master/test/com/hazelcast/qe/pipeline/distribution/dockerfiles/DockerfileUpdateStepTest.groovy#L70
cc @Holmistr

File: Dockerfile (hunk 2 above)
Comment (leszko) [reply Dockerfile]: Removed.

**Comment by mesutcelik** [comment]
Just one more thing to note that Size is not a big problem always. As Docker is layered, only moving part in this image is hazelcast-enterprise-all.jar section which is update in every hazelcast release. The rest will stay the same so will be cached in existing deployments.

**Comment by leszko** [comment]
Merging this PR and we can move the discussion (about merging `hazelcast-enterprise` and `hazelcast-openshift` images) outside of this thread.

**Pull request merged**